src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/SQLClause.java

```diff
@@ -91,4 +101,10 @@ private SQLMethodInvokeExpr wrapNestedFunction(SQLExpr expr) {
         return nestedFunc;
     }
 
+    String pathFromIdentifier(SQLExpr identifier) {
+        String field = Util.extendedToString(identifier);
+        int lastDot = field.lastIndexOf(".");
```

Comment:
those indexOfs have bitten us earlier. Are we 100% sure that field will always have a . in it?

Reply:
Though I am 99.9% sure we will atleast `.`, but let me rethink any scenario where this would not be the case.

Reply:
seems Chen's PR earlier had some special case with multi-field columns/fields.

Reply:
Keeping the logic same for know to match [existing behavior](https://github.com/opendistro-for-elasticsearch/sql/pull/98/files#diff-128e1606ce985235d9324d79ae94cddcR101) , this might break with inner fields in object type in nested docs. Will verify that special case and handle in a separate PR.
Created issue #142